### PR TITLE
AARA-395: fix for blank page on RFA list

### PR DESCRIPTION
**Pull request by LeonidMarushevskyi**
#### Which User story does this relate to (link)?
- https://osi-cwds.atlassian.net/browse/AARA-395

#### Brief feature description
- Rendering of RFA list was failing for objects which have empty address array in residence field. Example of residence object: 
{"addresses": [], "home_languages": [], "weapon_in_home": null, "directions_to_home": "", "body_of_water_exist": null, "residence_ownership": null, "physical_mailing_similar": null, "body_of_water_description": "", "others_using_residence_as_mailing": false, "other_people_using_residence_as_mailing": null}

#### Does this PR block/effect other stories?
- No

#### Do any commands needs to be run in order to run the project with this PR?
- No

#### Code coverage percent diff as a result of this PR?
- 88.24% Statements 
- 79.6% Branches 
-  85.79% Functions 
-  88.44% Lines

### CODE REVIEW

File: app/javascript/rfa_forms/rfa01a_list/applicationsListRow.js [hunk 1]
```diff
@@ -13,8 +13,8 @@ const getPhoneNumber = (phonesList) => {
 }
 
 const getAddress = (address, key) => {
-  const residentialAddress = address.get(0)
-  return residentialAddress.get(key)
+  const residentialAddress = address && address.get(0)
+  return residentialAddress && residentialAddress.get(key)
```
Comment (anuroop15) [inline app/javascript/rfa_forms/rfa01a_list/applicationsListRow.js]: good catch on address empty or not. But is residentialAddress check needed in return statement. As object will have street, city and zip in all cases.

File: test/javascript/rfa_forms/rfa01a_list/applicationsListRow_tests.js [hunk 2]
```diff
@@ -65,3 +65,24 @@ describe('Verify Application List Rows', () => {
...
+    facilityId: undefined,
+    familyName: undefined,
+    applicationStatus: undefined,
+    applicationReceivedDate: undefined,
+    applicantsInfo: application,
+    applicationAddress: address
+  }
+  const AppListViewCard = shallow(<ApplicationsListRow {...props} />)
+  it('To Return table Row', () => {
+    expect(AppListViewCard.find('tr').length).toEqual(1)
```
Comment (anuroop15) [inline test/javascript/rfa_forms/rfa01a_list/applicationsListRow_tests.js]: good to have check on table with table header. 
ecpect(AppListViewCard.find('th').at(0)).toBe('FacilityID'). to make sure table header is rendered.

File: app/javascript/rfa_forms/rfa01a_list/applicationsListRow.js (hunk 1 above)
Comment (LeonidMarushevskyi) [reply app/javascript/rfa_forms/rfa01a_list/applicationsListRow.js]: It is needed. If address is empty array then residentialAddress will be undefined. Then on line 17 we will have "Cannot read property 'get' of undefined". You can reproduce this error using test which I wrote.

Comment (anuroop15) [reply app/javascript/rfa_forms/rfa01a_list/applicationsListRow.js]: defaultProps should take of this scenario. Not sure why it's not working in this case.

File: test/javascript/rfa_forms/rfa01a_list/applicationsListRow_tests.js (hunk 2 above)
Comment (LeonidMarushevskyi) [reply test/javascript/rfa_forms/rfa01a_list/applicationsListRow_tests.js]: Added test to check that default address is used when address supplied as empty array

**Pull request merged**